flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -342,6 +346,14 @@ private ContainerLaunchContext createTaskExecutorLaunchContext(Resource resource
 		FiniteDuration teRegistrationTimeout = new FiniteDuration(timeout, TimeUnit.SECONDS);
 		final Configuration taskManagerConfig = BootstrapTools.generateTaskManagerConfiguration(
 				flinkConfig, "", 0, 1, teRegistrationTimeout);
+		// Add resource profile of slots to task executor config. 
+		// For yarn, all slots in a task executor have same resource profile 
+		ByteArrayOutputStream output = new ByteArrayOutputStream();
+		ObjectOutputStream rpOutput = new ObjectOutputStream(output);
+		rpOutput.writeObject(getResourceProfile(priority.getPriority()));
+		rpOutput.close();
+		taskManagerConfig.setString(ConfigConstants.TASK_MANAGER_RESOURCE_PROFILE_KEY,
+				new String(Base64.encodeBase64(output.toByteArray())));
```

Comment:
Thank @rmetzger for the explanation. I think maybe we can use flink-conf.yaml temporally and add a separate config file later.